### DISCUSSION TITLE
Implement CURLOPT_PROXYAUTH

### DISCUSSION THREAD

**Discussion by drzraf** [OpenIDC/mod_auth_openidc#1107] [Ideas]
In order to forward requests to the provider, `CURLOPT_PROXY` is already available via the `OIDCOutgoingProxy` configuration directive, but there is no way to allow requests to flow through an **authenticated** proxy. For this, `CURLOPT_PROXYUSERPWD` would need to be implemented.

Use-case: My servers hosted by a European hosting provider relies on GSuite OpenIDC provider. Google recently started blocking some of this competing hosting provider's network range from reaching some of its API endpoints (including oauth). As such I need to proxy Apache2's `mod_auth_openidc` requests through an (unblocked) network where an (**authenticated**) proxy server lives.

**Comment by zandbelt** [comment]
that would be a useful improvement indeed, a (tested) PR is welcome

**Comment by zandbelt** [comment]
would you be able to confirm that https://github.com/OpenIDC/mod_auth_openidc/commit/f276311c368de4d2bf11639535dcc09c96de2277 works for you?

**Reply by drzraf** [reply]
Wow, so quick!
Definitely worth a try, but I don't know when I'll be able to compile & test.

**Reply by zandbelt** [reply]
let us know for which platform you'd need a binary package and we'll make one available for you to test

**Reply by drzraf** [reply]
x86-64 on Apache 2.4.56 on Debian 11

```
linux-vdso.so.1
libcurl.so.4
libcjose.so.0
libjansson.so.4
libcrypto.so.1.1
libpcre.so.3
libhiredis.so.0.14
libc.so.6
libnghttp2.so.14
libidn2.so.0
librtmp.so.1
libssh2.so.1
libpsl.so.5
libssl.so.1.1
libgssapi_krb5.so.2
libldap_r-2.4.so.2
liblber-2.4.so.2
libbrotlidec.so.1
libz.so.1
libpthread.so.0
libdl.so.2
libunistring.so.2
libgnutls.so.30
libhogweed.so.6
libnettle.so.8
libgmp.so.10
libgcrypt.so.20
libkrb5.so.3
libk5crypto.so.3
libcom_err.so.2
libkrb5support.so.0
libresolv.so.2
libsasl2.so.2
libbrotlicommon.so.1
libp11-kit.so.0
libtasn1.so.6
libgpg-error.so.0
libkeyutils.so.1
libffi.so.7
```

**Reply by zandbelt** [reply]
please try https://www.mod-auth-openidc.org/downloads/libapache2-mod-auth-openidc_2.4.14.4rc4-1.bullseye_amd64.deb

**Reply by drzraf** [reply]
Wow @zandbelt : It works so well! and the satisfaction to finally bypass this awful restriction Google placed arbitrarily on our hosting provider is so enjoyable!

Thank you very much!
